starlette/exceptions.py

```diff
@@ -9,11 +9,14 @@
 
 
 class HTTPException(Exception):
-    def __init__(self, status_code: int, detail: str = None) -> None:
+    def __init__(
+        self, status_code: int, detail: str = None, headers: dict = None
```

Comment:
```suggestion
        self, status_code: int, detail: str = None, headers: typing.Optional[typing.Mapping[str, str]] = None
```

`dict` == `Dict[Any, Any]`, so let's start with the most restrictive version here and we can expand it later. Plus `Response` accepts typing.Mapping[str, str]` ([here](https://github.com/encode/starlette/blob/81bd557de69c1a780d81e94c1e6ee08009377052/starlette/responses.py#L59)) so it makes sense to match that.

Reply:
I don't think the `typing.Optional` is needed in the current situation.

Reply:
I've implemented with `typing.Mapping` initially, but when the type is more restrictive, there's a mypy issue, and `Response` accepts `headers: dict = None`, so I cannot add the right type here until `Response` is fixed.

Reply:
Ah okay, we should fix Response.
Optional isn't needed, but it's the right thing to do. Otherwise, MyPy strict will complain (see https://mypy.readthedocs.io/en/stable/command_line.html#none-and-optional-handling)

Reply:
I get that. I just wanted to say if we want to go for mypy strict, there are a lot of other changes needed.

For example in the same link `detail: str = None` needs to be `detail: typing.Optional[str] = None`. I just wanted to point that out. I've seen both cases in the code anyway.

Reply:
Oh yeah, a lot of other changes would be needed. But since using Optional doesn't hurt anything I just go with that as a default choice.

Reply:
Conceptually, it does hurt. :thinking: 

Using `Optional[<type>]`, you're saying that the parameter accepts `None`, with `<type>`, you're saying that it doesn't accept `None`... even if the default value is `None`.

Reply:
Agreed, but I also think that having a default value that's not accepted isn't very intuitive, I'd rather use `_NOT_SET = object()` or whatever. Anyhow, this is getting off topic, I think my request was answered in https://github.com/encode/starlette/pull/1435#discussion_r792699586 and it's not worth holding up this PR for typing related discussion